packages/app/src/navigation/SidebarNavigation.vue

```diff
@@ -98,20 +99,62 @@ const navigation = [
   { name: 'Settings', icon: SettingsIcon, href: '/settings' },
 ]
 
+gql`
+fragment SidebarNavigation on Query {
+  localSettings {
+    preferences {
+      isSideNavigationOpen
+      isSpecsListOpen
+      autoScrollingEnabled
+      reporterWidth
+      specListWidth
+    }
+  }
+}
+`
+
+gql`
+mutation SideBarNavigation_SetPreferences ($value: String!) {
+  setPreferences (value: $value) {
+    ...SidebarNavigation
+  }
+}`
+
 gql`
 query SideBarNavigation {
   ...SidebarNavigationHeader
+  ...SidebarNavigation
 }
 `
 
 const query = useQuery({ query: SideBarNavigationDocument, requestPolicy: 'network-only' })
 
-const bindingsOpen = ref(false)
+const setPreferences = useMutation(SideBarNavigation_SetPreferencesDocument)
 
-const mainStore = useMainStore()
+const bindingsOpen = ref(false)
 
 const route = useRoute()
 
 const navIsAlwaysCollapsed = computed(() => route.meta?.navBarExpandedAllowed !== false)
 
+const isNavBarExpanded = ref(true)
+
+watchEffect(() => {
+  if (route.meta?.navBarExpandedAllowed === false) {
+    isNavBarExpanded.value = false
+  } else {
+    isNavBarExpanded.value = query.data?.value?.localSettings.preferences.isSideNavigationOpen ?? true
+  }
+})
+
+const updatePreferences = async () => {
+  if (route.meta?.navBarExpandedAllowed === false) {
+    return
+  }
+
+  const resp = await setPreferences.executeMutation({ value: JSON.stringify({ isSideNavigationOpen: !isNavBarExpanded.value }) })
+
+  isNavBarExpanded.value = resp.data?.setPreferences?.localSettings.preferences.isSideNavigationOpen ?? true
```

Comment:
Do you think there's any value to optimistically updating `isNavBarExpanded.value` first, then executing the mutation? I know under normal conditions the mutation is fast, but maybe there are situations like high memory usage that could make this sluggish without needing to be.